plugins/module_utils/network/sonic/config/ssh/ssh.py

```diff
@@ -0,0 +1,352 @@
+        requests = []
+        state = self._module.params['state']
+
+        client_want = {}
+        client_have = {}
+        if want and want.get('client', None):
+            client_want['client'] = want['client']
+        if have and have.get('client', None):
+            client_have['client'] = have['client']
+        client_commands, client_requests = self.handle_ssh_client_replaced_overridden(client_want, client_have)
+        requests.extend(client_requests)
+
+        if client_commands and len(requests) > 0:
+            commands = update_states(client_commands, state)
+
+        return commands, requests
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        commands = []
+        requests = []
+
+        client_want = {}
+        client_have = {}
+        if want and want.get('client', None):
+            client_want['client'] = want['client']
+        if have and have.get('client', None):
+            client_have['client'] = have['client']
+        client_commands, client_requests = self.handle_ssh_client_merged(client_want, client_have)
+        requests.extend(client_requests)
+
+        if client_commands and len(requests) > 0:
+            commands = update_states(client_commands, 'merged')
+
+        return commands, requests
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+        requests = []
+
+        client_want = {}
+        client_have = {}
+        if want and want.get('client', None):
+            client_want['client'] = want['client']
+        if have and have.get('client', None):
+            client_have['client'] = have['client']
+        client_commands, client_requests = self.handle_ssh_client_deleted(client_want, client_have)
+        requests.extend(client_requests)
+
+        if client_commands and len(requests) > 0:
+            commands = update_states(client_commands, "deleted")
+
+        return commands, requests
+
+    def handle_ssh_client_replaced_overridden(self, want, have):
+        """Requests and commands necessary to migrate the current configuration
+           to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        if want:
+            diff = get_diff(have, want)
+            commands = diff
+        if commands:
+            requests = self.delete_specific_ssh_client_params(commands)
+            commands = update_states(commands, "deleted")
+
+        diff = get_diff(want, have)
+        if diff:
+            commands = diff
+            requests.extend(self.modify_specific_ssh_client_params(commands))
```

Comment:
This overwrites the "deleted" commands, so they don't show up in the output.

(Sorry we missed this earlier.)

Instead of this, please do the following (or something equivalent):

- Move the extraction of the "state" from _state_replaced_overridden to this function. (It will no longer be needed there.)

and then, here:

```suggestion
            mod_commands = diff
            mod_requests = self.modify_specific_ssh_client_params(mod_commands)
            if mod_commands and len(mod_requests) > 0:
                mod_commands = update_states(mod_commands, state)
                commands.extend(mod_commands)
                requests.extend(mod_requests)
```

Reply:
Addressed.